### ISSUE TITLE
configuration for interaction bipartite graph on recommendation system

### ISSUE THREAD

**Issue by suhmily** [facebookresearch/PyTorch-BigGraph#180]
May I ask if BigGraph supports multiple entity types like "LHS-userid" & "RHS-itemid"?  An example config file is provided  on movieLens would be better.
Thanks!

**Comment by lw** [comment]
Yes, PBG supports multiple entity types. The syntax for the config in such cases is like this:
```
entities={
    "LHS-userid": {"num_partitions": 1},
    "RHS-itemid": {"num_partitions": 1},
},
relations=[
    {"name": "follow", "lhs": "LHS-userid", "rhs": "RHS-itemid", "operator": "none"}
],
```
Unfortunately we don't have a full example readily available for this.

Note that you can have as many entity types as you want, and similarly as many relation types, which can use the entity types in all sorts of ways (e.g., entity types can be repeated, they can appear on both sides, ...).

**Comment by RunfengZhang** [comment]
Hi Luca, I also find PBG is extremely useful to train a large-scale CF model due to its excellent training throughputs. When the label is binary one, I think we can apply PBG directly on that dataset. But how about the case when training labels are non-negative real numbers? For instance, the rating scores between 0 and 5? Could we achieve that NMF by leveraging PBG? Thanks! @lw

**Comment by lw** [comment]
@adamlerer recently added support for edge weights, he can perhaps help.
